kingpin/actors/aws/elb.py

```diff
@@ -228,3 +216,60 @@ def _execute(self):
             yield utils.tornado_sleep(3)
 
         raise gen.Return()
+
+
+class UseCert(ELBBaseActor):
+
+    """Find a server cert in IAM and use it for a specified ELB."""
+
+    all_options = {
+        'name': (str, REQUIRED, 'Name of the ELB'),
+        'region': (str, REQUIRED, 'AWS region name, like us-west-2'),
+        'cert_name': (str, REQUIRED, 'Unique IAM certificate name, or ARN'),
+    }
+
+    def __init__(self, *args, **kwargs):
+        """Set up additional IAM connection."""
+        super(UseCert, self).__init__(*args, **kwargs)
+        self.iam_conn = boto.iam.connection.IAMConnection()
+
+    @concurrent.run_on_executor
+    @utils.exception_logger
+    def _find_cert(self, name):
+        """Return a boto IAM object for a certificate."""
+        self.log.debug('Searching for cert "%s"...' % name)
+        try:
+            cert = self.iam_conn.get_server_certificate(name)
+        except BotoServerError as e:
+            raise CertNotFound(
+                'Could not find cert %s. Reason: %s' % (name, e))
+        return cert
+
+    @concurrent.run_on_executor
+    @utils.exception_logger
+    def _use_cert(self, elb, cert):
+        """Assign an ssl cert to a given ELB.
+
+        Args:
+            elb: boto elb object
+            cert: boto iam server_certificate object
+        """
+
+        arn = cert['get_server_certificate_response'].get(
+            'get_server_certificate_result').get(
+            'server_certificate').get(
+            'server_certificate_metadata').get('arn')
+        self.log.info('Setting ELB "%s" to use cert arn: %s' % (elb, arn))
+        elb.set_listener_SSL_certificate(443, cert.arn)
+
+    @gen.coroutine
+    def _execute(self):
+        """Find ELB, and a Cert, then apply it."""
+        elb = yield self._find_elb(self.option('name'))
+        cert = yield self._find_cert(self.option('cert_name'))
+
+        if self._dry:
+            self.log.info('Would instruct %s to use %s' % (
```

Comment:
I think using ARN is a bad idea. Especially if we want to validate that this ARN is real -- too many useless API calls, without a benefit. I'll rewrite `_find_cert()` to be smarter and cache the result instead.